cmp/form/FormModel.ts

```diff
@@ -274,4 +308,83 @@ export class FormModel extends HoistModel {
             }
         );
     }
+
+    private initPersist({
+        fieldsToInclude = null,
+        fieldsToExclude = null,
+        path = 'formValues',
+        ...rootPersistWith
+    }: FormPersistOptions) {
+        const fieldNameMap = this.createFormFieldsToPersistListing(
+            fieldsToInclude,
+            fieldsToExclude
+        );
+
+        PersistenceProvider.create({
+            persistOptions: {
+                path,
+                ...rootPersistWith
+            },
+            target: {
+                getPersistableState: () => {
+                    return new PersistableState(
+                        fieldNameMap.reduce(
+                            (acc, name) => ({
+                                ...acc,
+                                [name]: this.serialize(this.fields[name].value)
+                            }),
+                            {}
+                        )
+                    );
+                },
+                setPersistableState: ({value: formValues}) => {
+                    // There is no metadata on a field to denote it is a date.
+                    // Use a regex matcher to tests for dates and format matches accurately.
+                    this.setValues(
+                        fieldNameMap.reduce((acc, name) => {
+                            return {...acc, [name]: this.deserialize(formValues[name])};
+                        }, {} as PlainObject)
+                    );
+                }
+            },
+            owner: this
+        });
+    }
+
+    private serialize(formValue: unknown) {
+        if (isArray(formValue)) return formValue.map(this.serialize);
```

Comment:
Does passing the function reference like this work? I think because `serialize` is not an arrow function, it will lose its "this" context when passed this way, but maybe it doesn't matter since the method implementation isn't referencing any other members on the model besides the method itself.

Reply:
Fixed